### ISSUE TITLE
Cant build TC from source

### ISSUE THREAD

**Issue by intelaravind** [facebookresearch/TensorComprehensions#430]
Hi,

I am trying build TC from source. I am following these instructions. 
https://facebookresearch.github.io/TensorComprehensions/installation_non_conda.html?highlight=install
and I get the following errors

```
[ 96%] Linking C executable isl_polyhedron_minimize
[ 97%] Linking C executable isl_polyhedron_sample
In file included from /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:146:0,
                 from /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_test.c:23:
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:144:12: **error: unknown type name ‘isl_pw_multi_aff_list’; use ‘struct’ keyword to refer to the type**
 #define EL isl_pw_multi_aff
            ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:3:19: note: in definition of macro ‘xLIST’
 #define xLIST(EL) EL ## _list
                   ^~
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:12: note: in expansion of macro ‘LIST’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
            ^~~~
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:17: note: in expansion of macro ‘EL’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
                 ^~
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:144:12: error: unknown type name ‘isl_pw_multi_aff_list’; did you mean ‘isl_pw_multi_aff’?
 #define EL isl_pw_multi_aff
            ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:3:19: note: in definition of macro ‘xLIST’
 #define xLIST(EL) EL ## _list
                   ^~
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:50: note: in expansion of macro ‘LIST’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
                                                  ^~~~
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:55: note: in expansion of macro ‘EL’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
```

**Environment and version details**

- ISL commit id 72af18e09781646a649b1b5aec9c593f24c78996 (https://github.com/nicolasvasilache/isl.git)
- TC commit id 6b319f5672ddc40a66a5d5d2e0481c8a1e7488bf 
- NVCC : 9.1.85
- GCC: tried with 5.3, 6.4 and 7.2 (all have the same error)
- OS: Ubuntu 18.04

**Comment by intelaravind** [comment]
Below is the command and the ouput with -H

`/usr/bin/gcc-5  -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build -I/home/aravind/anaconda3/include -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build/include -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/include  -H   -o CMakeFiles/isl_schedule.dir/schedule.c.o   -c /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/schedule.c
`
```

.. /usr/include/x86_64-linux-gnu/bits/wordsize.h
....... /home/aravind/anaconda3/include/isl/set_type.h
....... /usr/include/x86_64-linux-gnu/bits/byteswap-16.h
Multiple include guards may be useful for:
/home/aravind/anaconda3/include/isl/hmap.h
/home/aravind/anaconda3/include/isl/maybe_templ.h
/usr/include/assert.h
/usr/include/x86_64-linux-gnu/bits/byteswap-16.h
/usr/include/x86_64-linux-gnu/bits/byteswap.h
/usr/include/x86_64-linux-gnu/bits/endian.h
/usr/include/x86_64-linux-gnu/bits/select.h
/usr/include/x86_64-linux-gnu/bits/stdlib-float.h
/usr/include/x86_64-linux-gnu/bits/sys_errlist.h
/usr/include/x86_64-linux-gnu/bits/typesizes.h
/usr/include/x86_64-linux-gnu/bits/uintn-identity.h
/usr/include/x86_64-linux-gnu/bits/waitflags.h
/usr/include/x86_64-linux-gnu/bits/waitstatus.h
/usr/include/x86_64-linux-gnu/gnu/stubs-64.h
/usr/include/x86_64-linux-gnu/gnu/stubs.h
....... /usr/include/x86_64-linux-gnu/bits/uintn-identity.h
...... /usr/include/x86_64-linux-gnu/sys/select.h
....... /usr/include/x86_64-linux-gnu/bits/select.h
........ /usr/include/x86_64-linux-gnu/bits/wordsize.h
. /home/aravind/anaconda3/include/isl/polynomial.h
.. /home/aravind/anaconda3/include/isl/constraint.h
....... /usr/include/x86_64-linux-gnu/bits/types/sigset_t.h
........ /usr/include/x86_64-linux-gnu/bits/types/__sigset_t.h
... /home/aravind/anaconda3/include/isl/point.h
....... /usr/include/x86_64-linux-gnu/bits/types/struct_timeval.h
........ /home/aravind/anaconda3/include/isl/local_space.h
.. /usr/include/x86_64-linux-gnu/bits/types/struct_timespec.h
...... /usr/include/x86_64-linux-gnu/sys/sysmacros.h
....... /usr/include/x86_64-linux-gnu/bits/sysmacros.h
.. /home/aravind/anaconda3/include/isl/point.h
... /home/aravind/anaconda3/include/isl/stdint.h
.. /home/aravind/anaconda3/include/isl/local_space.h
...... /usr/include/x86_64-linux-gnu/bits/pthreadtypes.h
....... /usr/include/x86_64-linux-gnu/bits/thread-shared-types.h
...... /home/aravind/anaconda3/include/isl/stdint.h
.... /usr/include/x86_64-linux-gnu/bits/pthreadtypes-arch.h
......... /usr/include/x86_64-linux-gnu/bits/wordsize.h
..... /usr/include/alloca.h
...... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
.. /home/aravind/anaconda3/include/isl/map.h
... /home/aravind/anaconda3/include/isl/stdint.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_scan.h
. /home/aravind/anaconda3/include/isl/options.h
..... /usr/include/x86_64-linux-gnu/bits/stdlib-float.h
.... /home/aravind/anaconda3/include/isl/arg.h
..... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
... /home/aravind/anaconda3/include/isl/printer_type.h
.. /home/aravind/anaconda3/include/isl/set.h
... /home/aravind/anaconda3/include/isl/map_type.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_reordering.h
.. /home/aravind/anaconda3/include/isl/hash.h
... /home/aravind/anaconda3/include/isl/stdint.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_blk.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_int.h
.... /usr/include/string.h
..... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
..... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
... /home/aravind/anaconda3/include/isl/aff_type.h
..... /usr/include/x86_64-linux-gnu/bits/types/locale_t.h
...... /usr/include/x86_64-linux-gnu/bits/types/__locale_t.h
... /home/aravind/anaconda3/include/isl/mat.h
.... /home/aravind/anaconda3/include/isl/vec.h
....... /usr/include/strings.h
....... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
.. /home/aravind/anaconda3/include/isl/val.h
...... /home/aravind/anaconda3/include/isl/multi.h
....... /home/aravind/anaconda3/include/isl/space.h
........ /home/aravind/anaconda3/include/isl/id.h
......... /home/aravind/anaconda3/include/isl/stdint.h
.... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build/isl_config.h
..... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_config_post.h
...... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_int_gmp.h
........ /home/aravind/anaconda3/include/gmp.h
........ /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
... /usr/lib/gcc/x86_64-linux-gnu/5/include/stdint.h
.......... /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h
........ /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h
........ /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h
............. /usr/include/limits.h
.......... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
.. /usr/include/stdint.h
............... /usr/include/x86_64-linux-gnu/bits/posix1_lim.h
.................. /usr/include/x86_64-linux-gnu/bits/local_lim.h
 /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
............ /usr/include/linux/limits.h
............ /usr/include/x86_64-linux-gnu/bits/posix2_lim.h
.......... /usr/include/x86_64-linux-gnu/bits/wchar.h
............ /usr/include/x86_64-linux-gnu/bits/wordsize.h
............ /usr/include/x86_64-linux-gnu/bits/stdint-uintn.h
........ /home/aravind/anaconda3/include/isl/printer.h
....... /home/aravind/anaconda3/include/isl/set_type.h
Multiple include guards may be useful for:
/usr/include/x86_64-linux-gnu/bits/byteswap-16.h
/usr/include/x86_64-linux-gnu/bits/byteswap.h
/usr/include/x86_64-linux-gnu/bits/endian.h
/usr/include/x86_64-linux-gnu/bits/select.h
/usr/include/x86_64-linux-gnu/bits/stdlib-float.h
/usr/include/x86_64-linux-gnu/bits/sys_errlist.h
/usr/include/x86_64-linux-gnu/bits/typesizes.h
/usr/include/x86_64-linux-gnu/bits/uintn-identity.h
/usr/include/x86_64-linux-gnu/bits/waitflags.h
/usr/include/x86_64-linux-gnu/bits/waitstatus.h
/usr/include/x86_64-linux-gnu/gnu/stubs-64.h
/usr/include/x86_64-linux-gnu/gnu/stubs.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_map_private.h
.. /home/aravind/anaconda3/include/isl/set.h
... /home/aravind/anaconda3/include/isl/map_type.h
... /home/aravind/anaconda3/include/isl/aff_type.h
... /home/aravind/anaconda3/include/isl/point.h
... /home/aravind/anaconda3/include/isl/local_space.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
... /home/aravind/anaconda3/include/isl/mat.h
.... /home/aravind/anaconda3/include/isl/vec.h
Multiple include guards may be useful for:
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build/isl_config.h
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_config_post.h
/usr/include/assert.h
/usr/include/limits.h
/usr/include/x86_64-linux-gnu/bits/byteswap-16.h
/usr/include/x86_64-linux-gnu/bits/byteswap.h
/usr/include/x86_64-linux-gnu/bits/endian.h
/usr/include/x86_64-linux-gnu/bits/local_lim.h
/usr/include/x86_64-linux-gnu/bits/select.h
/usr/include/x86_64-linux-gnu/bits/stdlib-float.h
/usr/include/x86_64-linux-gnu/bits/sys_errlist.h
/usr/include/x86_64-linux-gnu/bits/typesizes.h
/usr/include/x86_64-linux-gnu/bits/uintn-identity.h
/usr/include/x86_64-linux-gnu/bits/waitflags.h
/usr/include/x86_64-linux-gnu/bits/waitstatus.h
/usr/include/x86_64-linux-gnu/gnu/stubs-64.h
/usr/include/x86_64-linux-gnu/gnu/stubs.h.
./usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h.
.. /home/aravind/anaconda3/include/isl/val.h
Multiple include guards may be useful for:
/usr/include/assert.h
/usr/include/x86_64-linux-gnu/bits/byteswap-16.h
/usr/include/x86_64-linux-gnu/bits/byteswap.h
/usr/include/x86_64-linux-gnu/bits/endian.h
/usr/include/x86_64-linux-gnu/bits/select.h
/usr/include/x86_64-linux-gnu/bits/stdlib-float.h
/usr/include/x86_64-linux-gnu/bits/sys_errlist.h
/usr/include/x86_64-linux-gnu/bits/typesizes.h
/usr/include/x86_64-linux-gnu/bits/uintn-identity.h
/usr/include/x86_64-linux-gnu/bits/waitflags.h
/usr/include/x86_64-linux-gnu/bits/waitstatus.h
/usr/include/x86_64-linux-gnu/gnu/stubs-64.h
/usr/include/x86_64-linux-gnu/gnu/stubs.h
...... /home/aravind/anaconda3/include/isl/multi.h
. /usr/include/stdlib.h
....... /home/aravind/anaconda3/include/isl/space.h
.. /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
... /usr/include/features.h
... /home/aravind/anaconda3/include/isl/stdint.h
.... /usr/include/x86_64-linux-gnu/sys/cdefs.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_equalities.h
... /usr/include/x86_64-linux-gnu/bits/wordsize.h
..... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_seq.h
... /home/aravind/anaconda3/include/isl/set_type.h
..... /usr/include/x86_64-linux-gnu/bits/long-double.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_scan.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_mat_private.h
.... /usr/include/x86_64-linux-gnu/gnu/stubs.h
..... /usr/include/x86_64-linux-gnu/gnu/stubs-64.h
.. /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_vec_private.h
.. /usr/include/x86_64-linux-gnu/bits/waitflags.h
.. /usr/include/x86_64-linux-gnu/bits/waitstatus.h
.. /usr/include/x86_64-linux-gnu/bits/floatn.h
... /usr/include/x86_64-linux-gnu/bits/floatn-common.h
.... /usr/include/x86_64-linux-gnu/bits/long-double.h
... /home/aravind/anaconda3/include/isl/point.h
.. /home/aravind/anaconda3/include/isl/map.h
... /home/aravind/anaconda3/include/isl/stdint.h
... /home/aravind/anaconda3/include/isl/local_space.h
... /home/aravind/anaconda3/include/isl/stdint.h
.. /usr/include/x86_64-linux-gnu/sys/types.h
... /usr/include/x86_64-linux-gnu/bits/types.h
.... /usr/include/x86_64-linux-gnu/bits/wordsize.h
.... /usr/include/x86_64-linux-gnu/bits/typesizes.h
... /usr/include/x86_64-linux-gnu/bits/types/clock_t.h
... /usr/include/x86_64-linux-gnu/bits/types/clockid_t.h
... /usr/include/x86_64-linux-gnu/bits/types/time_t.h
... /usr/include/x86_64-linux-gnu/bits/types/timer_t.h
... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
... /usr/include/x86_64-linux-gnu/bits/stdint-intn.h
.. /home/aravind/anaconda3/include/isl/map.h
... /home/aravind/anaconda3/include/isl/stdint.h
... /usr/include/endian.h
.... /usr/include/x86_64-linux-gnu/bits/endian.h
.... /usr/include/x86_64-linux-gnu/bits/byteswap.h
..... /usr/include/x86_64-linux-gnu/bits/wordsize.h
..... /usr/include/x86_64-linux-gnu/bits/byteswap-16.h
.... /usr/include/x86_64-linux-gnu/bits/uintn-identity.h
... /usr/include/x86_64-linux-gnu/sys/select.h
.... /usr/include/x86_64-linux-gnu/bits/select.h
..... /usr/include/x86_64-linux-gnu/bits/wordsize.h
.... /usr/include/x86_64-linux-gnu/bits/types/sigset_t.h
..... /usr/include/x86_64-linux-gnu/bits/types/__sigset_t.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_reordering.h
.... /usr/include/x86_64-linux-gnu/bits/types/struct_timeval.h
.. /home/aravind/anaconda3/include/isl/hash.h
... /home/aravind/anaconda3/include/isl/stdint.h
.... /usr/include/x86_64-linux-gnu/bits/types/struct_timespec.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_blk.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_int.h
... /usr/include/x86_64-linux-gnu/sys/sysmacros.h
.... /usr/include/string.h
..... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
.... /usr/include/x86_64-linux-gnu/bits/sysmacros.h
..... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
... /usr/include/x86_64-linux-gnu/bits/pthreadtypes.h
.... /usr/include/x86_64-linux-gnu/bits/thread-shared-types.h
..... /usr/include/x86_64-linux-gnu/bits/pthreadtypes-arch.h
...... /usr/include/x86_64-linux-gnu/bits/wordsize.h
..... /usr/include/x86_64-linux-gnu/bits/types/locale_t.h
...... /usr/include/x86_64-linux-gnu/bits/types/__locale_t.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_reordering.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
.. /usr/include/alloca.h
..... /usr/include/strings.h
... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
...... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
.... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build/isl_config.h
..... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_config_post.h
.... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_int_gmp.h
..... /home/aravind/anaconda3/include/gmp.h
...... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
.. /usr/include/x86_64-linux-gnu/bits/stdlib-float.h
. /home/aravind/anaconda3/include/isl/options.h
.. /home/aravind/anaconda3/include/isl/arg.h
... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
...... /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h
....... /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h
........ /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h
......... /usr/include/limits.h
. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h
.......... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
.. /home/aravind/anaconda3/include/isl/aff.h
... /home/aravind/anaconda3/include/isl/union_set_type.h
.... /home/aravind/anaconda3/include/isl/union_map_type.h
.......... /usr/include/x86_64-linux-gnu/bits/posix1_lim.h
........... /usr/include/x86_64-linux-gnu/bits/local_lim.h
.. /home/aravind/anaconda3/include/isl/ctx.h
............ /usr/include/linux/limits.h
... /usr/include/stdio.h
.... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
.......... /usr/include/x86_64-linux-gnu/bits/posix2_lim.h
.... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
.... /usr/include/x86_64-linux-gnu/bits/types/__FILE.h
.... /usr/include/x86_64-linux-gnu/bits/types/FILE.h
.... /usr/include/x86_64-linux-gnu/bits/libio.h
..... /usr/include/x86_64-linux-gnu/bits/_G_config.h
...... /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h
...... /usr/include/x86_64-linux-gnu/bits/types/__mbstate_t.h
..... /usr/lib/gcc/x86_64-linux-gnu/5/include/stdarg.h
.... /usr/include/x86_64-linux-gnu/bits/stdio_lim.h
.... /usr/include/x86_64-linux-gnu/bits/sys_errlist.h
. /home/aravind/anaconda3/include/isl/schedule.h
.. /home/aravind/anaconda3/include/isl/union_set_type.h
... /home/aravind/anaconda3/include/isl/union_map_type.h
.... /home/aravind/anaconda3/include/isl/list.h
..... /home/aravind/anaconda3/include/isl/printer_type.h
.. /home/aravind/anaconda3/include/isl/schedule_type.h
.. /home/aravind/anaconda3/include/isl/aff_type.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_pw_templ.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_pw_macro.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_pw_templ.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_pw_macro.h
.. /home/aravind/anaconda3/include/isl/space.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_multi_templ.h
... /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_multi_macro.h
... /home/aravind/anaconda3/include/isl/id.h
.... /home/aravind/anaconda3/include/isl/stdint.h
..... /usr/lib/gcc/x86_64-linux-gnu/5/include/stdint.h
.. /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h
...... /usr/include/stdint.h
....... /usr/include/x86_64-linux-gnu/bits/libc-header-start.h
....... /usr/include/x86_64-linux-gnu/bits/wchar.h
....... /usr/include/x86_64-linux-gnu/bits/wordsize.h
In file included from /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:146:0,
                 from /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_test.c:23:
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:144:12: error: unknown type name ‘isl_pw_multi_aff_list’
 #define EL isl_pw_multi_aff
            ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:3:19: note: in definition of macro ‘xLIST’
 #define xLIST(EL) EL ## _list
                   ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:12: note: in expansion of macro ‘LIST’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
            ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:17: note: in expansion of macro ‘EL’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
                 ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_aff_private.h:144:12: error: unknown type name ‘isl_pw_multi_aff_list’
 #define EL isl_pw_multi_aff
            ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:3:19: note: in definition of macro ‘xLIST’
 #define xLIST(EL) EL ## _list
                   ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:50: note: in expansion of macro ‘LIST’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
                                                  ^
/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/isl_list_templ.h:16:55: note: in expansion of macro ‘EL’
 __isl_give LIST(EL) *FN(LIST(EL),dup)(__isl_keep LIST(EL) *list);
                                                       ^

```

**Comment by skimo-openhub** [comment]
On Mon, May 21, 2018 at 10:45:16AM -0700, ASR wrote:
> Below is the command and the ouput with -H
> 
> `/usr/bin/gcc-5
> -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp
> -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build
> -I/home/aravind/anaconda3/include

Does this one come from GMP_INCLUDE_DIRS?

> -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/build/include
> -I/home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/include
> -H   -o CMakeFiles/isl_schedule.dir/schedule.c.o   -c
> /home/aravind/scratch/tc/TensorComprehensions/third-party/islpp/schedule.c

Try

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25e87a47e9..de434d2238 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -368,7 +368,7 @@ macro (add_isl_executable _name)
   set(_sources ${ARGN})
   add_executable("${_name}" ${_sources})
   set_target_properties("${_name}" PROPERTIES FOLDER "Integer Set Library")
-  target_include_directories("${_name}" PRIVATE "${ISL_SOURCE_DIR}" "${ISL_BINARY_DIR}" ${GMP_INCLUDE_DIRS} ${PIPLIB_INCLUDE_DIRS})
+  target_include_directories("${_name}" PRIVATE "${ISL_SOURCE_DIR}" "${ISL_BINARY_DIR}" ${PIPLIB_INCLUDE_DIRS})
   target_link_libraries("${_name}" isl)
 endmacro ()
 
> ... /home/aravind/anaconda3/include/isl/aff_type.h

This is clearly the wrong include.

skimo

**Comment by intelaravind** [comment]
yes /home/aravind/anaconda3/include came from GMP_INCLUDE_DIRS

I applied the above patch and its working now. 

Thanks

**Comment by skimo-openhub** [comment]
Should be fixed (indirectly) by #447.
